runtime/simx/vortex.cpp

```diff
@@ -415,6 +437,21 @@ class vx_device {
       return 1;
     }
 
+    // HW: virtual mem allocator has the same address range as global_mem. next step is to adjust it
+    virtual_mem_ = new MemoryAllocator(ALLOC_BASE_ADDR, (GLOBAL_MEM_SIZE - ALLOC_BASE_ADDR), MEM_PAGE_SIZE, CACHE_BLOCK_SIZE);
+    CHECK_ERR(virtual_mem_reserve(PAGE_TABLE_BASE_ADDR, (GLOBAL_MEM_SIZE - PAGE_TABLE_BASE_ADDR), VX_MEM_READ_WRITE), {
+      return err;
+    });
+    CHECK_ERR(virtual_mem_reserve(STARTUP_ADDR, 0x40000, VX_MEM_READ_WRITE), {
+      return err;
+    });
```

Comment:
This is to be removed from the memory range as well in the future?

Reply:
I'm not sure if I understand your question correctly, but Line 441 create the memory allocator for virtual memory, so it is independent from global (physical) memory allocator.